google/cloud/bigquery/_helpers.py

```diff
@@ -315,13 +315,21 @@ def _timestamp_to_json_parameter(value):
 def _timestamp_to_json_row(value):
     """Coerce 'value' to an JSON-compatible representation."""
     if isinstance(value, datetime.datetime):
+        # For naive datetime objects UTC timezone is assumed, thus we format
```

Comment:
Was this always our assumption? Based on the unit tests, I suppose so?

Reply:
Can't say if "always", but the `_datetime_to_json()` helper has been unchanged since 2017 and was supposedly doing its job fine with naive datetime objects in this time - at least I don't remember any reported issues related to timezones. The `strftime()` conversion just uses the `datetime` fields at their face value and ignores any `tzinfo` set.

Considering that naive datetimes are used most of the time (as opposed to timezone-aware datetime objects) and that users did not complain, the UTC timezone assumption was apparently correct behavior. The logic just failed to cover the case when somebody actually used a timezone-aware instance.

And indeed, unit tests expect the same, a naive `datetime` instance is supposed to be converted to a string representation with a zero offset from UTC ([example](https://github.com/googleapis/python-bigquery/blob/530e1e8d8fe8939e914a78ff1b220907c1b87af7/tests/unit/test__helpers.py#L729-L731)).